src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -126,7 +126,7 @@ std::string reorder_options(const std::string& org_options) {
 }
 
 inline bool does_options_support_batch_compilation(const std::string& options) {
-    return options.find("-D") == std::string::npos && options.find("-I") == std::string::npos;
+    return options.find("-I") == std::string::npos;
```

Comment:
@yeonbok Could you check this change? Do you think it would be valid? These options are created as follows:
```cpp
        kernel_string->options = exe_mode + " -cl-mad-enable";
        if (engine_info.bOptHintsSupport)
            kernel_string->options += " -DOPT_HINS_SUPPORTED=1";
        if (engine_info.bLocalBlockIOSupport)
            kernel_string->options += " -Dcl_intel_subgroup_local_block_io -DLOCAL_BLOCK_IO_SUPPORTED=1";
```
So far I see only these possible definitions, and all of them depend on device caps only and don't depend on specific kernel, thus batch compilation seems to be possible.

Reply:
Yeah you might remember I asked you about this thing. I also was curious why this condition is preventing batch compilation and it was in my backlog to check further. I walso wondering about "-I" because it seems not be able to be supported by compiler. So seems that we can get rid of this function itself.. how do you think?

Also, "OPT_HINS" seems to be a typo in the OV.

Reply:
@andrei-cv  please replace `OPT_HINS_SUPPORTED` with `OPT_HINTS_SUPPORTED` and get rid of `does_options_support_batch_compilation` function completely.